VMEncryption/main/Utils/HandlerUtil.py

```diff
@@ -403,72 +408,108 @@ def redo_current_status(self):
                               stat[0]["status"]["code"],
                               stat[0]["status"]["formattedMessage"]["message"])
 
+    def reject_settings(self):
+        try:
+            previous_seq = str(self.get_latest_seq() - 1)
+            previous_status_file = os.path.join(self._context._status_dir, previous_seq + '.status')
+            
+            stat_rept = waagent.GetFileContents(previous_status_file)
```

Comment:
If a previous status file is not found, it seems this would throw an exception which would be handled but then exit without writing a status file at the current sequence.  If the extension exits without writing a status file corresponding to the current sequence number, I think guest agent will report a QOS impacting failure for that particular operation.

Reply:
I will change the logic here to handle the scenario better.

Reply:
Done.